tests/permissions_validator_test.py

```diff
@@ -23,9 +37,66 @@ def test_permissions_artifact_name(self):
         container_mock = type(
             "FakeContainer",
             (),
-            {"request": request_mock, "create_bucket_storage": Mock(return_value=storage_mock), "logger": Mock()}
+            {"request": request_mock, "create_silent_bucket_storage": Mock(return_value=storage_mock), "logger": Mock()}
         )
         validator = PermissionsValidator(container_mock())
         self.asserts.json_equals(self.permissions_no_name, validator.permissions)
         storage_mock.get_artifact_as_string.assert_called_with("_keys/TOKEN")
         container_mock.logger.warning.assert_called_with("Name was not set in authorization token.")
+
+    def test_no_artifact_not_logged(self):
+        """
+            A pretty functional test. What I really wanted out of this
+            test was to make sure that the PermissionsValidator would use
+            the "silent_logger" for the storage that it would use.  We
+            don't want it to log a message with the authorization token
+            that was used. This potentially could be sensitive information.
+            For instance in the case where a user sends a valid token but
+            accidentally misspells the bucket name.
+        """
+        # Create a bucket but don't add anything to it. I only need it to
+        # NOT find the token I am requesting.
+        boto_connection = boto.connect_s3()
+        boto_connection.create_bucket("test")
+
+        # Use StdoutCatcher to gather the stdout for use later.
+        with StdoutCatcher() as catcher:
```

Comment:
This is slick. I would have done the same thing but rather then having an object that can be used in with context, I would have just replaced stdout right there and it would've looked stupid and everyone would LAUGH AT ME. WHY IS THIS WORLD SO CRUEL! Anyways cool solution.